pages/claimstatus/index.tsx

```diff
@@ -18,16 +35,87 @@ export default function Home(): ReactElement {
         <link rel="icon" href="/favicon.ico" />
       </Head>
-
       <Header />
       <Main />
       <Footer />
+      {console.log("-Test Response-")} {/* testing, to be removed before launch */}
+      {console.dir({claimData})} {/* testing */}
     </Container>
   )
 }
 
-export const getServerSideProps: GetServerSideProps = async ({ locale }) => ({
-  props: {
-    ...(await serverSideTranslations(locale || 'en', ['common', 'header', 'footer'])),
-  },
-})
+export const getServerSideProps: GetServerSideProps = async ({ req, locale }) => {
+  const PASSWORD = process.env.CERTIFICATE_PASSPHRASE
+  const API_URL: string = process.env.API_URL || "fail gracefully" //TODO: Fail gracefully. 
+  // const UNIQUE_NUMBER_HEADER: string = process.env.UNIQUE_NUMBER_HEADER_TITLE || "x-unique-number"
+  const CERT_DIR: (string) = (process.env.CERTIFICATE_DIR || "/var/ssl")
+  const KEY_PATH: string = path.join(CERT_DIR, '/ct.p12')
+
+  let apiData: (string | null) = null
+
+  // return certificate object with cert and key fields. 
+  async function getCertificate() {
+    const pemReadPkcs12 = promisify(pem.readPkcs12)
+    const pfx = fs.readFileSync(KEY_PATH);
+
+    const cert = await pemReadPkcs12(pfx, { p12Password: PASSWORD })
+
+    return cert;
+  }
+  
+  // TODO: Acquire uniqueNumber from header and use it to build API string 
+  // that will eventually be passed to makeRequest. 
+  // TODO: if no uniqueNumber, redirect. 
+  // const requestHeaders: IncomingHttpHeaders  = req.headers;
+  // const uniqueNumber = requestHeaders[UNIQUE_NUMBER_HEADER];
+  // function buildApiUrl(uniqueNumber, USER_KEY){}
+
+  async function makeRequest(certificate) {
+    const headers = {
+      Accept: 'application/json'
+    };
+
+    /* https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options
+    
+    // TODO: Store certs in memory, instead of loading every time. 
+    // TODO: rejectUnauthorized - set to true or implement a `checkServerIdentity` 
+    // function to check that the certificate is actually
+    //  issued by the host you're connecting to.
+    //  https://nodejs.org/api/https.html#https_https_request_url_options_callback
+    */
+    const options = {
+      cert: certificate.cert,
+      key: certificate.key,
+      rejectUnauthorized: false, 
+      keepAlive: false
+    };
+    const sslConfiguredAgent = new https.Agent(options);
+  
+    try {
+      const response = await fetch(API_URL, {
+        headers: headers, 
+        agent: sslConfiguredAgent, 
+      });
+  
+      const responseBody = await response.json();
+
+      apiData = responseBody
+
+    } catch (error) {
+      console.log(error);
+    }
+
+    return apiData;
+  }
+
+  // where it comes together
+  const certificate = await getCertificate()
+  const data = await makeRequest(certificate)
+
+  return ({
+    props: {
+      claimData: [data],
+      ...(await serverSideTranslations(locale || 'en', ['common', 'header', 'footer'])),
```

Comment:
just curious why was this necessary to add / is this something we do elsewhere?

Reply:
the new content here is the `claimData: [data]` - everything else is the existing app!